src/ServiceProvider.php

```diff
@@ -29,6 +29,9 @@ public function boot(): void
      */
     public function register(): void
     {
+        if (!config('elastic-apm-laravel.active')) {
```

Comment:
I'm afraid this will make the middleware fail because it expects the Agent to be defined.

Not registering the listeners should be enough. It will still create the transaction, but nothing else will be collected and also the agent will not send anything to Elastic. Let me make the change for you.